### PR TITLE
Synchronize from single-object endpoint

### PR DESCRIPTION
**Pull request by rjzondervan**

### CODE REVIEW

File: lib/Service/SynchronizationService.php [hunk 1]
```diff
@@ -174,6 +175,30 @@ private function getOriginId(Synchronization $synchronization, array $object): i
         return $originId;
     }
 
+	public function getObjectFromSource(Synchronization $synchronization, string $endpoint): array
```
Comment (WilcoLouwerse) [inline lib/Service/SynchronizationService.php:178]: Missing docblock

File: lib/Service/SynchronizationService.php [hunk 2]
```diff
@@ -383,6 +421,7 @@ public function getAllObjectsFromApi(Synchronization $synchronization, ?bool $is
         ];
 
         // Make the initial API call
+		// @TODO: method is now fixed to GET, but could end up in configuration.
```
Comment (WilcoLouwerse) [inline lib/Service/SynchronizationService.php:424]: Just a reminder that we should avoid doing this (even though I do it myself as well 😝)
Maybe create an issue for this, or mention it in another existing issue?

File: lib/Service/SynchronizationService.php [hunk 3]
```diff
@@ -174,6 +175,30 @@ private function getOriginId(Synchronization $synchronization, array $object): i
...
+			'headers' => $headers,
+			'query' => $query,
+		];
+
+		if (str_starts_with($endpoint, $source->getLocation()) === true) {
+			$endpoint = str_replace(search: $source->getLocation(), replace: '', subject: $endpoint);
+		}
+
+		// Make the initial API call
+		// @TODO: method is now fixed to GET, but could end up in configuration.
```
Comment (WilcoLouwerse) [inline lib/Service/SynchronizationService.php:196]: Just a reminder that we should avoid doing this (even though I do it myself as well 😝)
Maybe create an issue for this, or mention it in another existing issue?

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]

File: lib/Service/SynchronizationService.php (hunk 2 above)
Comment (rjzondervan) [reply lib/Service/SynchronizationService.php:435]: We should create an issue for this. I refined with @rubenvdlinde that I would put in these TODOs

Comment (WilcoLouwerse) [reply lib/Service/SynchronizationService.php:435]: Please show me the new issue created for this

Comment (rjzondervan) [reply lib/Service/SynchronizationService.php:435]: https://conduction.atlassian.net/browse/CONNECTOR-152?atlOrigin=eyJpIjoiZmEwYTA0NGE4NjlkNGEzODk3NTUxNmIzYTRlMTUwZDMiLCJwIjoiaiJ9

**Review by WilcoLouwerse — APPROVED** [review]

**Pull request merged**